lib/sanbase/clickhouse/labels.ex

```diff
@@ -0,0 +1,104 @@
+defmodule Sanbase.Clickhouse.Label do
+  @moduledoc """
+  Labeling transaction addresses
+  """
+
+  @type label :: %{
+          name: String.t(),
+          metadata: String.t()
+        }
+
+  @type input_transaction :: %{
+          from_address: %{
+            address: String.t(),
+            is_exhange: boolean
+          },
+          to_address: %{
+            address: String.t(),
+            is_exhange: boolean
+          },
+          trx_value: float,
+          trx_hash: String,
+          datetime: Datetime.t()
+        }
+
+  @type output_transaction :: %{
+          from_address: %{
+            address: String.t(),
+            is_exhange: boolean,
+            labels: list(label)
+          },
+          to_address: %{
+            address: String.t(),
+            is_exhange: boolean,
+            labels: list(label)
+          },
+          trx_value: float,
+          trx_hash: String,
+          datetime: Datetime.t()
+        }
+
+  @spec add_labels(list(input_transaction)) :: {:ok, list(output_transaction)}
+  def add_labels([]), do: {:ok, []}
+
+  def add_labels(transactions) when is_list(transactions) do
+    addresses = get_list_of_addresses(transactions)
+    {query, args} = addresses_labels_query(addresses)
+
+    Sanbase.ClickhouseRepo.query_reduce(query, args, %{}, fn [address, label, metadata], acc ->
+      metadata =
+        case Jason.decode(metadata) do
+          {:ok, _} -> metadata
+          {:error, reason} -> ~s|""|
+        end
+
+      Map.update(acc, address, [%{name: label, metadata: metadata}], fn labels ->
+        label = %{name: label, metadata: metadata}
+        labels ++ [label]
+      end)
```

Comment:
```suggestion
     label = %{name: label, metadata: metadata}
     Map.update(acc, address, [label], & [label | &1])
```
Is there any reason to append to the back? It is slow. If you need, just call `reverse` on the labels in the end, but there is no `datetime` to require a specific order.